### PR TITLE
Show integration usage

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

Show integration usage with the information gathered by the analytics integration.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [ ] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/javascripts/integration_usage.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+
+
+async function setUsage() {
+    const domain = document.querySelector("#integration_usage_script").getAttribute("domain")
+    const target_element = document.querySelector("#integration_usage")
+
+    const res = await fetch("https://analytics.home-assistant.io/data.json")
```
Comment (frenck) [inline source/javascripts/integration_usage.js]: Why resolve this in the browser end? We can also download this during build time and add it as a data file to Jekyll?

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/javascripts/integration_usage.js (hunk 1 above)
Comment (ludeeus) [reply source/javascripts/integration_usage.js]: it will do that, still testing

File: source/_includes/asides/component_navigation.html [hunk 2]
```diff
@@ -50,6 +50,13 @@
     <div class='section'>
       Source: <a href='https://github.com/home-assistant/core/tree/dev/homeassistant/components/{{ page.ha_domain }}'>View on GitHub</a>
     </div>
+    {%- if site.data.analytics_data.integrations[page.ha_domain] -%}
+      {% assign percentage = 100 | times: site.data.analytics_data.integrations[page.ha_domain] | divided_by: site.data.analytics_data.reports_integrations | ceil %}
+      <div class='section analytics-data'>
+        <span>Active in {{ percentage }}% Home Assistant installations<i>*</i></span>
+        <i>*Calculated by the <a href="/integrations/analytics">Analytics integration</a></i>
```
Comment (frenck) [inline source/_includes/asides/component_navigation.html]: Maybe we should link this to the big analytics page instead?
```suggestion
        <i>*Calculated by the <a href="/integrations/analytics">Analytics integration</a></i>
```

File: source/_includes/asides/component_navigation.html [hunk 3]
```diff
@@ -50,6 +50,13 @@
     <div class='section'>
       Source: <a href='https://github.com/home-assistant/core/tree/dev/homeassistant/components/{{ page.ha_domain }}'>View on GitHub</a>
     </div>
+    {%- if site.data.analytics_data.integrations[page.ha_domain] -%}
+      {% assign percentage = 100 | times: site.data.analytics_data.integrations[page.ha_domain] | divided_by: site.data.analytics_data.reports_integrations | ceil %}
+      <div class='section analytics-data'>
+        <span>Active in {{ percentage }}% Home Assistant installations<i>*</i></span>
```
Comment (frenck) [inline source/_includes/asides/component_navigation.html]:
```suggestion
        <span>Used by {{ percentage }}% of the Home Assistant installations<i>*</i></span>
```

File: source/_includes/asides/component_navigation.html (hunk 2 above)
Comment (frenck) [reply source/_includes/asides/component_navigation.html]: Alternatively, maybe not even mention it?

image

Just link the statement containing the percentage to the analytics page?

Comment (frenck) [reply source/_includes/asides/component_navigation.html]: Hmmmm that might look ugly too... Not sure myself 🤷

File: source/_includes/asides/component_navigation.html [hunk 4]
```diff
@@ -18,39 +18,22 @@
...
-    </div>
-  {%- endif -%}
-
-  {%- if page.ha_release -%}
-    <div class='section'>
-      Introduced in release: {{ page.ha_release }}
-    </div>
-  {%- endif -%}
+  <div class='section'>
+    The {{ page.name | default: page.title }} integration was added with the {{ page.ha_release | default: "0.7" }} release.
```
Comment (ludeeus) [inline source/_includes/asides/component_navigation.html]:
```suggestion
    The {{ page.name | default: page.title }} integration was added with the {{ page.ha_release | default: "unknown" }} release.
```
"unknown"  maybe, we will get feedback and can adjust if it's missing somewhere?

File: source/_includes/asides/component_navigation.html [hunk 5]
```diff
@@ -18,39 +18,22 @@
...
+    {% assign percentage = 100 | times: site.data.analytics_data.integrations[page.ha_domain] | divided_by: site.data.analytics_data.reports_integrations | ceil %}
+    <a title="Open analytics.home-assistant.io" href="https://analytics.home-assistant.io" target="_blank" rel="noopener">{{ percentage }}%</a> of Home Assistant installations is using this integration
+    {%- if page.ha_iot_class -%}
```
Comment (ludeeus) [inline source/_includes/asides/component_navigation.html]: default iOT class to "unknown"  maybe, we will get feedback and can adjust if it's missing somewhere?

File: source/_includes/asides/component_navigation.html [hunk 6]
```diff
@@ -19,19 +19,19 @@
...
     {% assign percentage = 100 | times: site.data.analytics_data.integrations[page.ha_domain] | divided_by: site.data.analytics_data.reports_integrations | ceil %}
-    <a title="Open analytics.home-assistant.io" href="https://analytics.home-assistant.io" target="_blank" rel="noopener">{{ percentage }}%</a> of Home Assistant installations is using this integration
-    {%- if page.ha_iot_class -%}
-    , it features a {{ page.ha_iot_class }} IoT class<sup><a href='/blog/2016/02/12/classifying-the-internet-of-things/#classifiers'><i class="icon-info-sign"></i></a></sup>
-    {%- endif -%}
+    {% assign iot_class = page.ha_iot_class | default: "unknown" %}
+
+    The {{ page.name | default: page.title }} integration was added in Home Assistant {{ page.ha_release | default: "unknown" }}.
+    and it's used by <a title="Open analytics.home-assistant.io" href="https://analytics.home-assistant.io" target="_blank" rel="noopener">{{ percentage }}%</a> of the active installations.
+    It's IoT class<sup><a href='/blog/2016/02/12/classifying-the-internet-of-things/#classifiers'><i class="icon-info-sign"></i></a></sup> is {{ iot_class }}
```
Comment (frenck) [inline source/_includes/asides/component_navigation.html]:
```suggestion
    Its IoT class<sup><a href='/blog/2016/02/12/classifying-the-internet-of-things/#classifiers'><i class="icon-info-sign"></i></a></sup> is {{ iot_class }}
```

File: source/_includes/asides/component_navigation.html [hunk 7]
```diff
@@ -19,19 +19,19 @@
...
 
     {% if page.ha_quality_scale -%}
-    {%- if page.ha_iot_class %} and a {%- else %} it features a {%- endif -%}
-      <a href='/docs/quality_scale/'> {{page.ha_quality_scale}} </a>quality scale
+      and scores {{page.ha_quality_scale}} on our <a href='/docs/quality_scale/'>quality scale.</a>
     {%- endif -%}
```
Comment (frenck) [inline source/_includes/asides/component_navigation.html]:
```suggestion
      and scores {{page.ha_quality_scale}} on our <a href='/docs/quality_scale/'>quality scale</a>
    {%- endif -%}.
```

**Review by frenck — APPROVED** [review]

**Comment by ludeeus** [comment]
:tada:

**Pull request merged**